server/api/utils/scheduler.py

```diff
@@ -944,6 +947,79 @@ def _enrich_schedule(
             labels.setdefault(mlrun_constants.MLRunInternalLabels.kind, fn_kind)
         return labels
 
+    @staticmethod
+    def _merge_schedule_and_schedule_object_labels(
+        labels: Optional[dict],
+        scheduled_object: Union[Optional[dict], Callable],
+    ):
+        """
+        Merges the labels of the scheduled object, giving precedence to the scheduled object labels
+        :param labels: the labels of a schedule
+        :param scheduled_object: a scheduled object
+
+        :return: merged labels
+        """
+        # Ensure scheduled_object is a dictionary-like object
+        if not hasattr(scheduled_object, "get"):
+            return labels
+
+        # Extract the scheduled object labels
+        scheduled_object_labels = (
+            scheduled_object.get("task", {}).get("metadata", {}).get("labels", {})
+        )
+
+        # If labels are empty, no need to update scheduled_object_labels,
+        if not labels:
+            return scheduled_object_labels
+
+        scheduled_object_labels = scheduled_object_labels or {}
+
+        # Merge labels, giving precedence to scheduled_object_labels
+        updated_labels = {**labels, **scheduled_object_labels}
+
+        # Update the original scheduled_object with the merged labels
+        scheduled_object.setdefault("task", {}).setdefault("metadata", {})["labels"] = (
+            updated_labels
+        )
+        return updated_labels
+
+    def _merge_schedule_and_db_schedule_labels(
```

Comment:
idk if its an official mlrun convention (as in if it was decided in development methodologies) but please move this above `_merge_schedule_and_schedule_object_labels` because this function uses `_merge_schedule_and_schedule_object_labels` and it makes it easier to read from top to bottom